### PR TITLE
[MJLINK-64] Fix addOptions arguments passing

### PR DESCRIPTION
**Pull request by XakepSDK**
Remove double quotes from --add-options values. With double quotes it treats arguments like -Dprop1=val1 -Dprop2=val2 as a single arg, so in runtime it prop2 is null and prop1 is val1 -Dprop2=val2

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MJLINK) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MJLINK-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MJLINK-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [x] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by bmarwell** [comment]
Thanks! Sorry, did not see this earlier. Can you rebase it please? I will take a look and merge it soon.

File: src/it/projects/cli-options/add-options/verify.groovy [hunk 1]
```diff
@@ -25,7 +25,7 @@ import org.codehaus.plexus.util.*
 
 String buildlog = new File( basedir, "build.log").text
 assert buildlog.contains( "--add-options" )
-assert buildlog.contains( "\"-Xmx128m --enable-preview -Dvar=value\"" )
+assert buildlog.contains( "-Xmx128m --enable-preview -Dvar=value" )
```
Comment (bmarwell) [inline src/it/projects/cli-options/add-options/verify.groovy:28]: To be extra safe, we could use this:

```groovy
assert buildlog.contains("\"-Xmx128m\" \"--enable-preview\" \"-Dvar=value\"")
```

WDYT?

File: src/main/java/org/apache/maven/plugins/jlink/JLinkMojo.java [hunk 2]
```diff
@@ -108,6 +108,8 @@
      *
      * <p>The command line equivalent is: {@code jlink --add-options="..."}.</p>
      *
+     * Spaces in options are not supported, they will be treated as separate options.
+     *
```
Comment (bmarwell) [inline src/main/java/org/apache/maven/plugins/jlink/JLinkMojo.java:112]: -1, this MUST be supported.

File: src/main/java/org/apache/maven/plugins/jlink/JLinkMojo.java [hunk 3]
```diff
@@ -654,7 +656,7 @@ private List<String> createJlinkArgs( Collection<String> pathsOfModules,
         if ( addOptions != null && !addOptions.isEmpty() )
         {
             jlinkArgs.add( "--add-options" );
-            jlinkArgs.add( String.format( "\"%s\"", String.join( " ", addOptions ) ) );
+            jlinkArgs.add( String.format( "%s", String.join( " ", addOptions ) ) );
```
Comment (bmarwell) [inline src/main/java/org/apache/maven/plugins/jlink/JLinkMojo.java:659]: We can support that here. We would just need to iterate over `addOptions` instead, so -1 on this one.

**Review by bmarwell — CHANGES_REQUESTED** [review]

File: src/main/java/org/apache/maven/plugins/jlink/JLinkMojo.java (hunk 3 above)
Comment (XakepSDK) [reply src/main/java/org/apache/maven/plugins/jlink/JLinkMojo.java:659]: I think I did this because jlink can't handle it properly

**Comment by XakepSDK** [comment]
Hi! Thank you for reviewing! I will rebase this when I have free time.

**Comment by bmarwell** [comment]
@slachiewicz  if you don't mind I will create a new PR for this soon...

**Comment by bmarwell** [comment]
Closed in favour of https://github.com/apache/maven-jlink-plugin/pull/166

**Pull request closed without merging**